addons/web/static/src/views/fields/copy_clipboard/copy_button.js

```diff
@@ -1,33 +1,54 @@
 /** @odoo-module **/
 
-const { Component, useEffect, useState } = owl;
+import { browser } from "@web/core/browser/browser";
+import { Tooltip } from "@web/core/tooltip/tooltip";
+import { useService } from "@web/core/utils/hooks";
 
+const { Component, useRef } = owl;
 export class CopyButton extends Component {
     setup() {
-        this.state = useState({
-            isCopied: false,
-        });
-        useEffect(
-            () => {
-                return () => clearTimeout(this.timeoutId);
-            },
-            () => [this.state.isCopied]
+        this.button = useRef("button");
+        this.popover = useService("popover");
+    }
+
+    showTooltip() {
+        const closeTooltip = this.popover.add(
+            this.button.el,
+            Tooltip,
+            { tooltip: this.props.successText },
         );
+        browser.setTimeout(() => {
+            closeTooltip();
+        }, 800)
     }
 
     async onClick() {
+        let canWrite;
+        try {
+            // some browsers don't fully support the Permissions API,
+            // we must handle this try/catch to avoid a crash while
+            // verifying if we can copy to clipboard
+            const { state } = await browser.navigator.permissions.query({
+                name: "clipboard-write"
+            });
+            canWrite = state !== "denied";
+        } catch {
+            canWrite = Boolean(browser.navigator.clipboard);
+        }
+        if (!canWrite) {
+            return console.info("Browser doesn't allow to copy to clipboard");
+        }
         // any kind of content can be copied into the clipboard using
         // the appropriate native methods
         if (typeof this.props.content === "string") {
-            navigator.clipboard.writeText(this.props.content).then(() => {
-                this.state.isCopied = true;
+            browser.navigator.clipboard.writeText(this.props.content).then(() => {
```

Comment:
probably needs a  catch for the case where the permission type was "prompt" and the user denied it

Reply:
I think we can safely remove the usage of the Permissions API.
From the specs (still unsure - only Chromium supports this API on Clipboard), clipboard-write is automatically granted whenever the page loads in a secure environment.
The usage of the API is mainly available for the clipboard-read, which is not the point of this component.

I will handle and catch errors from the copy differently.